jakarta-package-alignment/src/main/java/com/palantir/gradle/jakartapackagealignment/JakartaPackageAlignmentPlugin.java

```diff
@@ -15,24 +15,55 @@
  */
 package com.palantir.gradle.jakartapackagealignment;
 
+import com.google.common.collect.ImmutableSet;
+import java.util.Set;
 import org.gradle.api.Plugin;
 import org.gradle.api.Project;
+import org.gradle.api.artifacts.Configuration;
 import org.gradle.api.artifacts.component.ModuleComponentSelector;
+import org.gradle.api.tasks.SourceSet;
+import org.gradle.api.tasks.SourceSetContainer;
 
 public class JakartaPackageAlignmentPlugin implements Plugin<Project> {
     @Override
     public final void apply(Project project) {
-        project.getConfigurations().configureEach(configuration -> {
-            configuration.getResolutionStrategy().getDependencySubstitution().all(dep -> {
-                if (dep.getRequested() instanceof ModuleComponentSelector) {
-                    ModuleComponentSelector selector = (ModuleComponentSelector) dep.getRequested();
-                    VersionMappings.getReplacement(selector)
-                            .ifPresent(replacement -> dep.useTarget(
-                                    replacement,
-                                    "forced to Java EE 8 dependency because the requested Jakarta "
-                                            + "dependency is < Jakarta EE 9"));
-                }
+        project.getPluginManager().withPlugin("java-base", _plugin -> {
+            SourceSetContainer sourceSets = project.getExtensions().getByType(SourceSetContainer.class);
+            sourceSets.configureEach(sourceSet -> {
+                configureAllConfigurationsForSourceSet(project, sourceSet);
             });
         });
+
+        project.getPluginManager().withPlugin("com.palantir.consistent-versions", _plugin -> {
+            // necessary to get g-c-v to write substituted dependencies to the versions.lock file
+            project.getConfigurations().getByName("unifiedClasspath", this::configureConfigurationForSubstitution);
```

Comment:
I wonder if we need something along these lines depending on the order things are applied?

(I defer to @CRogers on this sort of thing, though)
```gradle
project.getConfigurations().configureEach(new Action<Configuration>() {
    @Override
    public void execute(Configuration config) {
        if ("unifiedClasspath".equals(config.getName())) {
            configureConfigurationForSubstitution(config);
        }
    }
});
```

Reply:
hm, i guess? `withPlugin` guarantees that the action won't run until after the plugin is applied, and this configuration is created when the plugin is applied: https://github.com/palantir/gradle-consistent-versions/blob/develop/src/main/java/com/palantir/gradle/versions/VersionsLockPlugin.java#L220-L226